### PR TITLE
Hw 5 versioning training & model card

### PR DESCRIPTION
**Pull request by strangerin**

### CODE REVIEW

File: experiments/model_cards/toxicity_card.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import model_card_toolkit as mct
+from datetime import date
+from model_card_toolkit import ModelCardToolkit, Owner, Reference, Risk, Limitation, UseCase, User
+import uuid
+
+# Initialize the Model Card Toolkit with a path to store generated assets
+output_dir = "./experiments/model_cards/model_card_assets"
+toolkit = ModelCardToolkit(output_dir=output_dir)
```
Comment (truskovskiyk) [inline experiments/model_cards/toxicity_card.py:8]: remove hardcode and wrap as python funtion please

File: experiments/models/toxicity_classifier/train.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import random
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+from transformers import AutoTokenizer, XLMRobertaConfig
+import wandb
+
+from utils import load_train_set, load_test_set, build_dataset, build_classifier, build_classifier_v2, train_classifier, predict_with_classifier, show_roc_auc
+
+# Set seeds for reproducibility
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/train.py:12]: could split this code into funtions please?

File: experiments/models/toxicity_classifier/utils.py [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+from tensorflow.keras.callbacks import Callback
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.metrics import roc_auc_score, roc_curve
+import tensorflow as tf
+import tensorflow_addons as tfa
+from transformers import TFXLMRobertaModel, XLMRobertaConfig
+from transformers import AutoTokenizer, XLMRobertaTokenizer
+# binary classification overfits like there is no tommorow
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/utils.py:20]: lol, what is this comment here for?

File: experiments/models/toxicity_classifier/utils.py [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+                        else:
+                            cur_lang = "en"
+                        max_proba = 0.0
+                        for cur_field_idx in indices_of_sentiment_fields:
+                            try:
+                                cur_proba = float(row[cur_field_idx])
+                            except:
+                                cur_proba = -1.0
+                            err_msg2 = err_msg + " Value {0} is wrong!".format(
+                                row[cur_field_idx]
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/utils.py:148]: this looks like a really hard code to read

File: experiments/models/toxicity_classifier/utils.py [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+                            id_value = int(row[id_field_index])
+                        except:
+                            id_value = -1
+                        err_msg2 = err_msg + " {0} is wrong ID!".format(
+                            row[id_field_index]
+                        )
+                        assert id_value >= 0, err_msg2
+                        text = row[text_field_index].strip()
+                        assert len(text) > 0, err_msg + " Text is empty!"
+                        if lang_field_index >= 0:
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/utils.py:208]: same, hard to read. could you ask ChatGPT to refactor it?

File: experiments/models/toxicity_classifier/utils.py [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+        plt.plot(
+            list(range(len(history.history["val_" + metric_name]))),
+            history.history["val_" + metric_name],
+            label="Validation {0}".format(metric_name),
+        )
+    plt.xlabel("Epochs")
+    plt.ylabel(metric_name)
+    plt.title("Training process")
+    plt.legend(loc="best")
+    plt.show()
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/utils.py:467]: might log this into WANDB too

File: experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/conda-environment.yaml [hunk 7]
```diff
@@ -0,0 +1,301 @@
+name: base
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/conda-environment.yaml:1]: please prune this, keep only used deps

File: experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/conda-environment.yaml [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+      - wcwidth==0.2.10
+      - webencodings==0.5.1
+      - websockets==12.0
+      - werkzeug==3.0.1
+      - wheel==0.40.0
+      - wrapt==1.14.1
+      - xmljson==0.2.0
+      - yarl==1.9.2
+      - zc-lockfile==3.0.post1
+prefix: /home/krabli/mambaforge
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/conda-environment.yaml:301]: remove hardcode

File: experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,253 @@
+absl-py==2.0.0
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163542-65knpcqp/files/requirements.txt:1]: remove wandb folder please

File: experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/conda-environment.yaml [hunk 10]
```diff
@@ -0,0 +1,302 @@
+name: base
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/conda-environment.yaml:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/requirements.txt [hunk 11]
```diff
@@ -0,0 +1,254 @@
+absl-py==2.0.0
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/requirements.txt:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/wandb-metadata.json [hunk 12]
```diff
@@ -0,0 +1,208 @@
+{
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163646-3e7egihd/files/wandb-metadata.json:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/conda-environment.yaml [hunk 13]
```diff
@@ -0,0 +1,302 @@
+name: base
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/conda-environment.yaml:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/requirements.txt [hunk 14]
```diff
@@ -0,0 +1,254 @@
+absl-py==2.0.0
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/requirements.txt:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/wandb-metadata.json [hunk 15]
```diff
@@ -0,0 +1,208 @@
+{
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_163746-g8aemnc3/files/wandb-metadata.json:1]: remove

File: experiments/models/toxicity_classifier/wandb/run-20231116_164434-1t0dstcq/files/conda-environment.yaml [hunk 16]
```diff
@@ -0,0 +1,302 @@
+name: base
```
Comment (truskovskiyk) [inline experiments/models/toxicity_classifier/wandb/run-20231116_164434-1t0dstcq/files/conda-environment.yaml:1]: remove

**Review by truskovskiyk — COMMENTED** [review]
see comments, please remove wandb folder from git

File: experiments/models/toxicity_classifier/utils.py (hunk 3 above)
Comment (strangerin) [reply experiments/models/toxicity_classifier/utils.py:20]: Jarring caffeine-induced frustration.

**Pull request closed without merging**